src/expect_interface.rs

```diff
@@ -218,3 +218,31 @@ impl<'a> ExpectGetProperty<'a> {
         self.tester
     }
 }
+
+pub struct ExpectDefineMetric<'a> {
+    tester: &'a mut Tester,
+    metric_type: Option<i32>,
+    name: Option<&'static str>,
+}
+
+impl<'a> ExpectDefineMetric<'a> {
+    pub fn expecting(
+        tester: &'a mut Tester,
+        metric_type: Option<i32>,
+        name: Option<&'static str>,
+    ) -> ExpectDefineMetric<'a> {
```

Comment:
Sorry... this can be `Self<'a>` too if I'm not wrong. 
The advantage of using `Self` over the explicit struct name is that it avoids unnecessary changes when refactoring the type to a new name in the future.

Reply:
But [as you can read for yourself](https://users.rust-lang.org/t/idiomatic-use-of-self/73754) there are people on both sides... My comment wasn't clear by "just" selecting the both lines. Anyways, just ignore this one, as I think you'd need yet another review to merge 🤦 merge queue fun!